src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,221 @@
+                // feed version set for the deployment.
+                FeedVersion latestFeedVersion = currentDeploymentFeedVersion.feedSource.retrieveLatest();
+                // Make sure the latest feed version is not going to supersede a pinned feed version.
+                if (pinnedFeedVersions
+                    .stream()
+                    .anyMatch(feedVersion -> feedVersion.feedSource.id.equals(latestFeedVersion.feedSourceId))
+                ) {
+                    String message = String.format("Feed version %s from feed source %s could not be advanced because a previous feed version has been pinned for project %s.",
+                        latestFeedVersion.id,
+                        latestFeedVersion.parentFeedSource().name,
+                        project.name);
+                    LOG.warn(message);
+                    continue;
+                }
+
+                // Update to the latest feed version.
+                updatedFeedVersionIds.add(latestFeedVersion.id);
+
+                // Throttle this auto-deployment if needed. For projects that haven't yet been auto-deployed, don't
+                // wait and go ahead with the auto-deployment. But if the project has been auto-deployed before and
+                // if the latest feed version was created before the last time the project was auto-deployed and
+                // there are currently-active jobs that could result in an updated feed version being created, then
+                // this auto deployment should be throttled.
+                if (
+                    project.lastAutoDeploy != null &&
+                        latestFeedVersion.dateCreated.before(project.lastAutoDeploy) &&
+                        currentDeploymentFeedVersion.feedSource.hasJobsInProgress()
+                ) {
+                    // Another job exists that should result in the creation of a new feed version which should then
+                    // trigger an additional AutoDeploy job.
+                    LOG.warn(
+                        "Feed source {} contains an active job that should result in the creation of a new feed version. Auto deployment will be skipped until that version has fully processed.",
+                        currentDeploymentFeedVersion.feedSource.name
+                    );
+                    shouldWaitForNewFeedVersions = true;
+                }
+
+                // Make sure the latest feed version has no critical errors.
+                if (latestFeedVersion.hasCriticalErrors()) {
+                    latestVersionsWithCriticalErrors.add(latestFeedVersion);
+                }
+            }
+
+            // Always update the deployment's feed version IDs with the latest feed versions which have not been pinned
+            // to a particular version.
+            deployment.feedVersionIds = updatedFeedVersionIds;
+            Persistence.deployments.replace(deployment.id, deployment);
```

Comment:
It might make sense to move this block to just after all of the checks have cleared. That way the deployment's feed versions don't get updated until a deployment will actually happen.

Reply:
Updated.